docs/man-pages/condor_token_create.rst

```diff
@@ -0,0 +1,127 @@
+      
+
+*condor_token_create*
+======================
+
+given a password file, create an authentication token for the TOKEN authentication method
+:index:`condor_token_create<single: condor_token_create; HTCondor commands>`\ :index:`condor_token_create command`
+
+Synopsis
+--------
+
+**condor_token_create** **-identity** *user@domain* [**-keyid** *keyid*]
+[**-authz** *authz* ...] [**-lifetime** *value*]
+[**-token** *filename*] [**-debug**]
+
+**condor_token_create** [**-help** ]
+
+Description
+-----------
+
+*condor_token_create* will read an HTCondor password file inside the
+*SEC_PASSWORD_DIRECTORY* (by default, this is the pool password) and use it to create an authentication token.
+The authentication token may be subsequently used by clients to authenticate
+against a remote HTCondor server.  Tokens allow fine-grained authentication
+as individual HTCondor users as opposed to pool password, where anything
+in possession of the pool password will authenticate as the same user.
+
+An identity must be specified for the token; this will be the client's
+resulting identity at the remote HTCondor server.
+If the **-lifetime** or (one or more) **-authz** options are specified,
+the token will contain additional restrictions that limit what the
+client will be authorized to do.
+
+If successful, the resulting token will be sent to ``stdout``; by specifying
+the **-token** option, it will instead be written to the user's token directory.
```

Comment:
1) That would be cool to add information about question how to make a token for the daemon-to-daemon authentication. How can we use `SEC_TOKEN_SYSTEM_DIRECTORY`? In the case of Windows?

2) Suppose that on the first machine (Unix with master, collector, negotiator and schedd), we generated a token for `tester@windows-qa`. On the second `windows-qa` Windows machine, a STARTD demon is deployed under the user `tester` (administrator). The first Unix machine does not see the second (it is behind the firewall), a CCB mechanism is used.
The user wants the EXECUTE node to be added to the pool. Do the user need to transfer token to the Windows machine? Where to put this token?

3) I would like to briefly learn the following from this document: how can these tokens be used and why should they be written to files? And why should these files be hidden from attackers?

Reply:
@hozblok - thanks for the feedback!

1.  If a compatible token is present in the system's token directory, it will be used from daemon-to-daemon.  Apologies - I do not know anything about how to set this up on Windows (would love to hear the answers to your questions though!).  That might be a good question for the HTCondor user's list.  @johnkn - have you tried the TOKEN workflow on Windows?
2.  If created with `condor_token_create`, the user could transfer the token to the Windows machine by hand (scp, save it on USB, print it out), install it in the appropriate directory, and enable TOKEN auth (default in 8.9.4).  _However_, rather than potentially-insecurely transferring the token, it is better to use `condor_token_request` (or use the fact the startd should automatically request tokens) from `windows-qa` and have the admin approve the token request.  The `condor_token_request` CLI will securely transfer things.
3.  I think the full security model is probably a bit much for the manpages.  I will try to address as much as I can in the next revision.

Great input!

Reply:
I think I addressed these as best I can.  Can you take a look?

Reply:
@bbockelm Thanks! It became much better. Special thanks for the detailed answers, I removed some of the gaps in my understanding and.. I have sent additional questions to the HTCondor user's list.

`I think the full security model is probably a bit much for the manpages.` I agree with you.